client/client.go

```diff
@@ -22,57 +22,334 @@ limitations under the License.
 package client
 
 import (
+	"errors"
 	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
 
 	"github.com/fsouza/go-dockerclient"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/config"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/fs"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/network"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/wrapper"
 	"github.com/opencontainers/runc/libcontainer/cgroups"
 )
 
-const endpoint string = "unix:///var/run/docker.sock"
+const (
+	endpoint         string = "unix:///var/run/docker.sock"
+	dockerVersionKey string = "Version"
+)
 
-type ContainerInfo struct {
-	Id      string
-	Status  string
-	Created int64
-	Image   string
+// DockerClientInterface provides methods i.a. for interaction with the docker API.
+type DockerClientInterface interface {
+	ListContainersAsMap() (map[string]docker.APIContainers, error)
+	GetStatsFromContainer(string, bool) (*wrapper.Statistics, error)
+	InspectContainer(string) (*docker.Container, error)
+	FindCgroupMountpoint(string) (string, error)
 }
 
-type DockerClientInterface interface {
-	ListContainers() ([]ContainerInfo, error)
-	FindCgroupMountpoint(subsystem string) (string, error)
+// DockerClient
```

Comment:
ok, added sth like below:
```
// DockerClient holds fsouza go-dockerclient instance ready for communication with the server endpoint `unix:///var/run/docker.sock`,
// cache instance which is used to store output from docker container inspect (to avoid execute inspect request multiply times, it is called only once per container)
// and diskUsageCollector which is responsible for collecting container disk usage (based on `du -u` command) in the background
```